waspc/cli/src/Wasp/Cli/Command/CreateNewProject/Templates.hs

```diff
@@ -0,0 +1,79 @@
+module Wasp.Cli.Command.CreateNewProject.Templates
+  ( getStarterTemplates,
+    StarterTemplateNames,
+    templatesToList,
+    isValidTemplateName,
+  )
+where
+
+import Control.Exception (try)
+import Data.Aeson (FromJSON, parseJSON, withObject, (.:))
+import qualified Network.HTTP.Simple as HTTP
+
+data StarterTemplateNames = StarterTemplateNames [String]
+
+getStarterTemplates :: IO (Maybe StarterTemplateNames)
```

Comment:
I would document that Nothing means we failed to fetch templates.
Or, you might go a bit different, and let it throw exceptions, and then in the caller you can decide to drop them all if you want.
Also, would rename to `getStarterTemplatesNames`.

Reply:
Maybe we can do something like `data StarterTemplateNames = StarterTemplateNames [String] | NoTemplateNamesAvailable` and be explicit about the failure to fetch? 

I'm not sure if that's an overkill.

Reply:
I've introduced a `StarterTemplateNamesFetchResult` type which can either be success or failure. Then it's the same as `Either` but with extra meaning.